gammapy/datasets/core.py

```diff
@@ -237,12 +237,19 @@ def contributes_to_stat(self):
         return np.array(contributions)
 
     def stat_sum(self):
-        """Compute joint statistic function value."""
-        stat_sum = 0
-        # TODO: add parallel evaluation of likelihoods
-        for dataset in self:
-            stat_sum += dataset.stat_sum()
-        return stat_sum
+        """Total statistic given the current model parameters and priors."""
+        prior_stat_sum = 0.0
+        if self.models is not None:
+            prior_stat_sum = self.models.parameters.prior_stat_sum()
+        return self._stat_sum_likelihood() + prior_stat_sum
+
+    def _stat_sum_likelihood(self):
+        """Total statistic given the current model parameters without the priors."""
+        stat = self.stat_array()
+
+        if self.mask is not None:
+            stat = stat[self.mask.data]
+        return np.sum(stat, dtype=np.float64)
```

Comment:
`stat_sum` should stay the same, and add
```
     def _stat_sum_likelihood(self):
         """Total statistic given the current model parameters without the priors."""
        stat_sum = 0
        for dataset in self:
            stat_sum += dataset. _stat_sum_likelihood()
        return stat_sum
```

Reply:
indeed my bad, a bad copy/paste
fixed now